### DISCUSSION TITLE
Clarification on sign()

### DISCUSSION THREAD

**Discussion by Fallenprogram** [Part-II-Computational-Physics/.github#84] [General]
Hi, this is probably a very trivial question, for expressing sign() function, is the python translation of that function copysign() or numpy.sign()? Because from what I am reading on the documentation and elsewhere, they don't seem to be equivalent and produce different results.

**Comment by ceciliegl** [comment]
Hi! This is a good question. The difference between the two functions seems to be the evaluation of sign(0), which is +1 for copysign and 0 for numpy.sign. It seems to me that the implementation used by the authors in the Lion papers uses a function similar to numpy.sign. I don't think the result will depend too much on which function you choose, as the argument of sign rarely is exactly zero.

**Reply by Fallenprogram** [reply]
Thank you very much for that input on the difference, as I can't find this, at least explicitly, in the Internet. However, I managed to figure it out to be numpy because like what you said, I had a look at their research paper in a bit more detail, and yes, they seem to implement it via numpy.sign.
